parsing_input.c

```diff
@@ -0,0 +1,61 @@
+#include "shell.h"
+/**
+ * calculate_tokens - function to calculate token command
+ * @lineptr: pointer to string
+ * @delim: delimiter between strings
+ * Return: number of tokens
+ */
+int calculate_tokens(char *lineptr, char *delim)
+{
+	int counter = 0;
+	char *token, *line_copy;
+
+	line_copy = _strcpy(line_copy, lineptr);
```

Comment:
You called your custom defined strcpy function without line_cpy pointing to an allocated memory in heap. strcpy doesn't malloc a memory space so you need to malloc a memory space before calling the function, otherwise you can use strdup or any custom defined strdup function you have created